### PR TITLE
Plugin doc

### PR DESCRIPTION
**Pull request by vgunning**
### Change log

Add in details for the plugin developer guide in the docs. Created a _docs/plugins-developer-guide.md

Mentions the plugin yaml spec file, argument so the plugin, and that the plugin shouldn't be concerned about the driver.

### CODE REVIEW

File: _docs/plugins-developer-guide.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+---
+layout: page
+title: Plugins Developer Guide
+permalink: /docs/plugins-developer-guide/
+---
+
+## Plugins Overview
+
```
Comment (mikeynap) [inline _docs/plugins-developer-guide.md]: Note that the driver should not matter to the plugin writer.

File: _docs/plugins-developer-guide.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+Plugins are intended to be generic tasks that are executed with either the `deploy`, `update`, `status`, or `destroy`. The task can use any driver that the systems supports with the preferred driver being a `docker` container. Each instantiation of a plugin is a service which uses the JSON from `/var/lib/sigma/input` for inputs and JSON written to `/var/lib/sigma/output` to designate outputs.
+
+Specifics: Anything that is printed to the standard out and standard error is something reported back as a log in a job output. Additionally, the service can write progress to the `/var/lib/sigma/progress` as a way to state what step the installation may be at or may have just completed. The exit status of the command indicates the success or failure of the command to the system.
```
Comment (mikeynap) [inline _docs/plugins-developer-guide.md]: Make a note that the input provided will match the spec of the plugin.yml file. Idk how to word it, but yeah.

File: _docs/plugins-developer-guide.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
+---
+layout: page
+title: Plugins Developer Guide
+permalink: /docs/plugins-developer-guide/
+---
+
+## Plugins Overview
+
+Plugins are intended to be generic tasks that are executed with either the `deploy`, `update`, `status`, or `destroy`. The task can use any driver that the systems supports with the preferred driver being a `docker` container. Each instantiation of a plugin is a service which uses the JSON from `/var/lib/sigma/input` for inputs and JSON written to `/var/lib/sigma/output` to designate outputs.
```
Comment (mikeynap) [inline _docs/plugins-developer-guide.md]: Note that the 'deploy' 'update', 'status' will be available as the first and only command line argument provided to the plugin (args[1]).

Comment (adamschaub) [inline _docs/plugins-developer-guide.md]: Agreed, might be beneficial to note exactly how a command is being passed to the container. I.e. prescribe that whatever script/binary to be run should be the entry point, and the container's cmd will be the verb.

**Comment by stephenrlouie** [comment]
LGTM

File: _docs/plugins-developer-guide.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
+---
+layout: page
+title: Plugins Developer Guide
+permalink: /docs/plugins-developer-guide/
+---
+
+## Plugins Overview
+
+Plugins are intended to be generic tasks that are executed with either the `deploy`, `update`, `status`, or `destroy`. The plugin will be executed using a driver that the systems supports with the preferred driver being a `docker` container. The plugin shouldn't have to be concerned with which driver is executing it. Each instantiation of a plugin is a service which uses the JSON from `/var/lib/sigma/input` for inputs and JSON written to `/var/lib/sigma/output` to designate outputs. The driver calls the plugin providing only one command line argument being either `deploy`, `update`, `status`, or `destroy`.
```
Comment (meganokeefe) [inline _docs/plugins-developer-guide.md]: replace "the systems" with "Sigma"

Comment (meganokeefe) [inline _docs/plugins-developer-guide.md]: remove "being either" and replace with ":"

File: _docs/plugins-developer-guide.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+When creating the plugin, it will consist of the script or executable, and the plugin specifications file based on [Plugins and Services](https://sigma-dev.github.io/sigma/docs/plugins-services/). This plugin defines how the inputs are handled and what outputs it generates. Individual services are then instantiations of the plugin with provided inputs.
+
+Specifics: Anything that is printed to the standard out and standard error is something reported back as a log in a job output. Additionally, the service can write progress to the `/var/lib/sigma/progress` as a way to state what step the installation may be at or may have just completed. The exit status of the command indicates the success or failure of the command to the system. The inputs provided will match the specifications from the plugin's yaml file. The outputs file written should also match the plugin's yaml file.
```
Comment (meganokeefe) [inline _docs/plugins-developer-guide.md]: remove "something" from "something reported back"

Comment (meganokeefe) [inline _docs/plugins-developer-guide.md]: replace "may be at" and "may have just" with "is at" and "has just completed"

File: _docs/plugins-developer-guide.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+## Update
+
+The `update` command should take the provided inputs and update the service.
+
+## Status
+
+The `status` command should verify that the service was set up correctly from the provided input. This could be querying endpoints, or any other way to verify that the service is functional. A non-zero exit status would indicate a failing status and a zero exit code would indicate that the service has been set up and is running correctly.
+
+## Destroy
+
```
Comment (meganokeefe) [inline _docs/plugins-developer-guide.md]: destory -> destroy

**Pull request merged**